core/Period/Factory.php

```diff
@@ -168,10 +168,10 @@ public static function makePeriodFromQueryParams($timezone, $period, $date)
         } else {
             if (!($date instanceof Date)) {
                 if (preg_match('/^(now|today|yesterday|yesterdaySameTime|last[ -]?(?:week|month|year))$/i', $date)) {
-                    $date = Date::factory($date, $timezone);
-                } else {
-                    $date = Date::factory($date);
+                    // calculate the date taking the current time into account
+                    $date = Date::factory(Date::factory($date)->setTime(date('H:i:s', Date::getNowTimestamp()))->getTimestamp(), $timezone)->getDatetime();
```

Comment:
Can we split this to multiple lines so it's clearer what's being done? Otherwise the change looks good.

Reply:
done